c7n/filters/core.py

```diff
@@ -346,11 +346,12 @@ def __call__(self, r):
 
     def process_set(self, resources, event):
         rtype_id = self.get_resource_type_id()
-        resource_map = {r[rtype_id]: r for r in resources}
+        compiled = jmespath.compile(rtype_id)
```

Comment:
jmespath expression evaluation is pretty slow if we're entering engine execution o(n) resource cardinality, I'd like to understand if we're going to get a performance regression here.

Reply:
alternative would be to only due this if '.' in rtype_id

Reply:
good point re '.' in rtype_id